src/frontends/pytorch/src/op/avg_poolnd.cpp

```diff
@@ -8,19 +8,23 @@
 #include "openvino/op/concat.hpp"
 #include "openvino/op/constant.hpp"
 #include "openvino/op/pad.hpp"
-#include "openvino/op/subtract.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/add.hpp"
 #include "utils.hpp"
 
+
 namespace ov {
 namespace frontend {
 namespace pytorch {
 namespace op {
 
 using namespace ov::op;
 
-OutputVector translate_avg_poolnd(const NodeContext& context) {
+OutputVector translate_avg_pool_base(const NodeContext& context, const Output<Node>& input) {
```

Comment:
Could you reduce code duplication by providing number of dimensions to base function and resolve it here instead of separately in each function?